Nkerman tutorial edits

nkerman opened the pull request:
Suggested additions to 2024 Python training.

\cc @maxinelasp

nkerman committed:
Nat's initial additions to python training nb 2024

nkerman committed:
Add bit on comments

maxineofficial commented:
Just going to stick all my comments in here because GitHub doesn't like Jupiter. 

1. I like your intro, but I actually think it should be in the slide deck instead of here
2. Section on comments is great
3. I think the bit on pointers should be removed - although it's interesting it barely ever comes up in Python. 
4. Good call on adding f-strings especially because I use some later lol
5. The error section on functions and methods should go directly under the first section on functions
6. Thank you for beefing up the plotting section, it's way better now

Overall looks great, thanks so much for adding this!

nkerman committed:
Address Maxine's comments

nkerman commented:
> Just going to stick all my comments in here because GitHub doesn't like Jupiter.
> 
> 1. I like your intro, but I actually think it should be in the slide deck instead of here
> 2. Section on comments is great
> 3. I think the bit on pointers should be removed - although it's interesting it barely ever comes up in Python.
> 4. Good call on adding f-strings especially because I use some later lol
> 5. The error section on functions and methods should go directly under the first section on functions
> 6. Thank you for beefing up the plotting section, it's way better now
> 
> Overall looks great, thanks so much for adding this!

Thanks Maxine! I addressed the changes you requested here in 1, 3, and 5. 
I've added the intro stuff to a "WHY to learn Programming" slide. I kept it after your HOW to learn programming, since I really like that as the very first intro slide, focusing on the meta skills
